deltacat/utils/pyarrow.py

```diff
@@ -211,6 +211,26 @@ def _get_kwargs(self, content_type: str, kwargs: Dict[str, Any]) -> Dict[str, An
         return kwargs
 
 
+def _create_s3_file_system(s3_client_kwargs: dict) -> s3fs.S3FileSystem:
+    if not s3_client_kwargs:
+        return s3fs.S3FileSystem(anon=True)
+
+    config_kwargs = {}
+    if s3_client_kwargs.get("config") is not None:
+        boto_config = s3_client_kwargs.pop("config")
+        for key, val in boto_config.__dict__.items():
```

Comment:
Minor: Do we need to fetch `.items()` from `__dict__`?

Reply:
yeah, if we need both key and values. Right?

Reply:
Isn't `.items()` directly accessible from `boto_config` if it's a dict type?

Reply:
Config is not a dict type. [Ref](https://github.com/ray-project/ray/blob/e464bf07af9f6513bf71156d1226885dde7a8f46/python/ray/autoscaler/_private/aws/config.py#L59-L61)